### PR TITLE
Online score submission fixes

### LINKED ISSUE #32: persist online scoreboard
it's in memory right now. Write to file so it's not wiped every update

### PR DESCRIPTION
**Pull request by cmavelis**
https://github.com/Aveli-games/flappy-fish/assets/16199008/53823035-5c81-434e-bf7d-bac46197156c

client
- now always submit a score
- added a signal to global scores script that the leaderboard UI will use to update async

server
- persist scores to .json file to updates won't wipe scores (closes #32 )

### CODE REVIEW

File: autoloads/local_high_scores.gd [hunk 1]
```diff
@@ -24,13 +24,7 @@ func load():
 		file.close()
 
 		if typeof(data) == TYPE_ARRAY:
-			# for backward compat, only needed for cam & tris once.
-			# Replace if/else with just scores=data afterward
-			if data.size() > 0 && data[0].has("Name"):
-				scores = _convert_score_file(data)
-				save()
-			else:
-				scores = data
```
Comment (cmavelis) [inline autoloads/local_high_scores.gd:33]: @avelitj12 am I right in assuming nobody else has played the game locally? and that you've converted your local file?

File: autoloads/local_high_scores.gd [hunk 2]
```diff
@@ -62,17 +56,3 @@ func submit_score(initials, score):
 		scores.insert(rank, {"name": initials, "score": score})
 		save()
 		return true
-
-# for backward compat, only needed for cam & tris once
```
Comment (cmavelis) [inline autoloads/local_high_scores.gd:66]: see above

File: autoloads/local_high_scores.gd (hunk 1 above)
Comment (avelitj12) [reply autoloads/local_high_scores.gd:33]: I can delete the local save files on mine and Chyanne's computers

**Review by avelitj12 — APPROVED** [review]
Merged in main. Overall I'm fine with always submitting a score and like these changes, though that means some of the logic for https://github.com/Aveli-games/flappy-fish/issues/29 will have to be different, which is fine. I'll post notes in that issue

**Pull request merged**